compiler/rustc_parse/src/parser/attr_wrapper.rs

```diff
@@ -218,24 +218,24 @@ impl<'a> Parser<'a> {
         let start_token = (self.token.clone(), self.token_spacing);
         let cursor_snapshot = self.token_cursor.clone();
         let start_pos = self.num_bump_calls;
-
         let has_outer_attrs = !attrs.attrs.is_empty();
-        let prev_capturing = std::mem::replace(&mut self.capture_state.capturing, Capturing::Yes);
         let replace_ranges_start = self.capture_state.replace_ranges.len();
 
-        let ret = f(self, attrs.attrs);
-
-        self.capture_state.capturing = prev_capturing;
-
-        let (mut ret, trailing) = ret?;
+        let (mut ret, trailing) = {
+            let prev_capturing =
+                std::mem::replace(&mut self.capture_state.capturing, Capturing::Yes);
```

Comment:
```suggestion
                mem::replace(&mut self.capture_state.capturing, Capturing::Yes);
```
If you get to this PR again before it's merged, could you import `std::{iter, mem}` into this file to avoid 3-segment paths?